src/test/s3/commands/uploadFile.test.ts

```diff
@@ -5,47 +5,186 @@
 
 import * as assert from 'assert'
 import * as vscode from 'vscode'
-import { FileSizeBytes, uploadFileCommand } from '../../../s3/commands/uploadFile'
-import { S3BucketNode } from '../../../s3/explorer/s3BucketNode'
+import { S3 } from 'aws-sdk'
+import { FileSizeBytes, getFileToUpload, promptUserForBucket, uploadFileCommand } from '../../../s3/commands/uploadFile'
 import { S3Node } from '../../../s3/explorer/s3Nodes'
+import { S3BucketNode } from '../../../s3/explorer/s3BucketNode'
 import { S3Client } from '../../../shared/clients/s3Client'
 import { MockOutputChannel } from '../../mockOutputChannel'
-import { FakeCommands } from '../../shared/vscode/fakeCommands'
 import { FakeWindow } from '../../shared/vscode/fakeWindow'
-import { anything, mock, instance, when, capture, verify } from '../../utilities/mockito'
+import { anything, mock, instance, when, capture } from '../../utilities/mockito'
+import { Commands } from '../../../shared/vscode/commands'
+import { Window } from '../../../shared/vscode/window'
+import { FakeCommands } from '../../shared/vscode/fakeCommands'
+
 
 describe('uploadFileCommand', function () {
     const bucketName = 'bucket-name'
     const key = 'file.jpg'
     const sizeBytes = 16
     const fileLocation = vscode.Uri.file('/file.jpg')
     const statFile: FileSizeBytes = _file => sizeBytes
-
+    let outputChannel: MockOutputChannel
     let s3: S3Client
-    let node: S3BucketNode
-
+    let bucketNode: S3BucketNode
+    let window: FakeWindow
+    let getBucket: (s3client:S3Client, window?: Window) => Promise<S3.Bucket | string >
+    let getFile: (document?: vscode.Uri, window?: Window) => Promise<vscode.Uri | undefined >
+    let commands: Commands
+    
     beforeEach(function () {
         s3 = mock()
-        node = new S3BucketNode(
+        commands = new FakeCommands()
+        bucketNode = new S3BucketNode(
             { name: bucketName, region: 'region', arn: 'arn' },
             new S3Node(instance(s3)),
             instance(s3)
         )
+        window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
+        outputChannel = new MockOutputChannel()
+        
+    })
+
+    describe('when it gets the node parameter', async function() {
+        this.beforeEach(function () {
+            s3 = mock()
+            commands = new FakeCommands()
+            bucketNode = new S3BucketNode(
+                { name: bucketName, region: 'region', arn: 'arn' },
+                new S3Node(instance(s3)),
+                instance(s3)
+            )
+            window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
+            outputChannel = new MockOutputChannel()
+        })
+       
+        it("asks for the file location, then uploads, display success message if aplicable", async function() {
```

Comment:
`'uploads successfully'`.
* Message is shorter
* Since there's no success without this sequence happening, no clarity is lost.
* Nit: re: `display success message if aplicable`, the tests should be absolute. Is there a separate codepath that would result in a success with no success message displayed? If so, there should be a test for it. If not, there's no reason to mention "if".